src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/converter/rulemanager/OfFlowModConverter.java

```diff
@@ -0,0 +1,241 @@
+        for (OFFlowStatsEntry entry : statsReply.getEntries()) {
+            FlowSpeakerData speakerData = FlowSpeakerData.builder()
+                    .cookie(new Cookie(entry.getCookie().getValue()))
+                    .flags(convertToRuleManagerFlags(entry.getFlags()))
+                    .priority(entry.getPriority())
+                    .table(OfTable.fromInt(entry.getTableId().getValue()))
+                    .match(OfMatchConverter.INSTANCE.convertToRuleManagerMatch(entry.getMatch()))
+                    .instructions(
+                            OfInstructionsConverter.INSTANCE.convertToRuleManagerInstructions(entry.getInstructions()))
+                    .build();
+            flowSpeakerData.add(speakerData);
+        }
+        return flowSpeakerData;
+    }
+
+    /**
+     * Convert stats reply.
+     */
+    public List<GroupSpeakerData> convertToGroupSpeakerCommandData(OFGroupDescStatsReply statsReply) {
+        List<GroupSpeakerData> commandData = new ArrayList<>();
+        for (OFGroupDescStatsEntry entry : statsReply.getEntries()) {
+        }
+        return commandData;
+    }
+
+    /**
+     * Convert group command data into OfGroupMod representation.
+     */
+    public OFGroupAdd convertInstallGroupCommand(GroupSpeakerData commandData, OFFactory ofFactory) {
+        List<OFBucket> buckets = commandData.getBuckets().stream()
+                .map(x -> toOfBucket(ofFactory, x)).collect(Collectors.toList());
+        return ofFactory.buildGroupAdd()
+                .setGroup(OFGroup.of((int) commandData.getGroupId().getValue()))
+                .setGroupType(toOfGroupType(commandData.getType()))
+                .setBuckets(buckets)
+                .build();
+
+    }
+
+    /**
+     * Convert group command data into OfGroupMod representation.
+     */
+    public OFGroupDelete convertDeleteGroupCommand(GroupSpeakerData commandData, OFFactory ofFactory) {
+        return ofFactory.buildGroupDelete()
+                .setGroup(OFGroup.of((int) commandData.getGroupId().getValue()))
+                .setGroupType(toOfGroupType(commandData.getType()))
+                .build();
+    }
+
+    private OFBucket toOfBucket(OFFactory ofFactory, Bucket bucket) {
+        List<OFAction> ofActions = OfInstructionsConverter.INSTANCE.convertApplyActions(bucket.getWriteActions(),
+                ofFactory);
+        Builder builder = ofFactory.buildBucket()
+                .setActions(ofActions)
+                .setWatchGroup(toOfGroup(bucket.getWatchGroup()));
+        if (bucket.getWatchPort() != null) {
+            builder.setWatchPort(toOfWatchPort(bucket.getWatchPort()));
+        }
+        return builder.build();
+    }
+
+    private OFPort toOfWatchPort(WatchPort watchPort) {
+        switch (watchPort) {
+            case ANY:
+                return OFPort.ANY;
+            default:
+                throw new IllegalArgumentException("Unknown watch port");
```

Comment:
Add details to error message. `String.format("Unknown watch port %s", watchPort)`